### PR TITLE
Uploading a potential fine-tuning notebook

### LINKED ISSUE #21: Continue Fine-Tuning CodeT5 with R Bioinformatics Dataset and Evaluate Effectiveness
Continue with the fine-tuning process of the CodeT5 model using the expanded dataset of R bioinformatics code from sources like Bioconductor, tutorials, and DataScience Meta.

Document all the processes about the effectiveness of the process and what can be improved

Acceptance Criteria:
- Complete the next phase of fine-tuning using the updated dataset.
- Document observations, improvements, and areas needing more data or adjustment.

### PR DESCRIPTION
**Pull request by oam67**
Fixes #21

**What was changed?**

The notebook for fine-tuning the CodeT5 model was created and modified to support masked code prediction for R code. Key components include loading and preprocessing a dataset of R code, configuring the CodeT5 tokenizer and model, setting up the training loop using Hugging Face's Trainer, and adding logic for evaluating and testing masked code completion.

**Why was it changed?**

The goal is to adapt the general purpose CodeT5 model for better performance on R code, specifically for tasks involving masked span prediction. The base model is not specialized for R syntax or idioms, so this fine-tuning step helps the model learn patterns specific to R, improving its usefulness in Bioinformatics code completion and repair scenarios.

**How was it changed?**

The file Fine_tune_CodeT5_for_generating_R_code.ipynb was created for this issue. It includes dataset loading logic for R source code files. It has preprocessing that masks spans of code for training, following the span corruption objective used by CodeT5. Tokenizer and model setup is done using Salesforce/codet5-base. The training loop is defined using Seq2SeqTrainer, with appropriate training arguments. Evaluation logic is setup using BLEU. Then finally there is an inference section for running the fine-tuned model on masked R code samples. 

**Screenshots that show the changes (if applicable):**
image

### CODE REVIEW

File: Fine Tuning/Fine_tune_CodeT5_for_generating_R_code.ipynb [hunk 1]
```diff
@@ -0,0 +1,10094 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training_args = TrainingArguments(\n",
+    "    output_dir=\"./codet5-r-generation\",\n",
+    "    num_train_epochs=3,\n",
+    "    per_device_train_batch_size=4,\n",
+    "    per_device_eval_batch_size=4,\n",
+    "    warmup_steps=50,\n",
+    "    weight_decay=0.01,\n",
```
Comment (Omp06) [inline Fine Tuning/Fine_tune_CodeT5_for_generating_R_code.ipynb:98]: Is this the appropriate weight decay and value for the training arguments? Have you tested any of these ?

File: Fine Tuning/Fine_tune_CodeT5_for_generating_R_code.ipynb [hunk 2]
```diff
@@ -0,0 +1,10094 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f8cedd4d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def generate_r_code(prompt):\n",
+    "    inputs = tokenizer(prompt, return_tensors=\"pt\", truncation=True).input_ids\n",
+    "    outputs = model.generate(inputs, max_length=256)\n",
```
Comment (Omp06) [inline Fine Tuning/Fine_tune_CodeT5_for_generating_R_code.ipynb:145]: I don't think it is smart to have a max length on the outputs. I feel like some information might be left out.

**Review by AjithAkuthota23 — APPROVED** [review]
